File: vdp-python/visa/test/visadirect/test_watch_list_screening.py
from visa.helpers.abstract_visa_api_client import AbstractVisaAPIClient
import json
import unittest
'''
@author: visa
'''

class TestWatchListScreening(unittest.TestCase):

    def setUp(self):
        self.abstract_visa_api_client = AbstractVisaAPIClient()
        self.watch_list_inquiry = json.loads(''' {
                  "acquirerCountryCode": "840",
                  "acquiringBin": "408999",
                  "address": {
                      "city": "Bangalore",
                      "cardIssuerCountryCode": "IND"
                  },
                  "referenceNumber": "430000367618",
                  "name": "Mohammed Qasim"
                  }''')
    
    def test_watch_list_inquiry(self):
        base_uri = 'visadirect/'
        resource_path = 'watchlistscreening/v1/watchlistinquiry'
        response = self.abstract_visa_api_client.do_mutual_auth_request(base_uri + resource_path, self.watch_list_inquiry, 'Watch List Inquiry Test','post')
        self.assertEqual(str(response.status_code) ,"200" ,"Watch List Inquiry test failed")
        pass

File: vdp-java/src/test/java/com/visa/vdp/visadirect/TestWatchListScreening.java
package com.visa.vdp.visadirect;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;

import java.util.HashMap;

import org.apache.http.HttpStatus;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import com.visa.vdp.utils.AbstractVisaAPIClient;
import com.visa.vdp.utils.MethodTypes;

public class TestWatchListScreening {
	
	String watchListInquiry;
	AbstractVisaAPIClient abstractVisaAPIClient;
	
	@BeforeTest(groups = "visadirect")
	public void setup() {
		this.abstractVisaAPIClient = new AbstractVisaAPIClient();
		this.watchListInquiry = 
			"{"
			     + "\"acquirerCountryCode\": \"840\","
				 + "\"acquiringBin\": \"408999\","
				 + "\"address\": {"
				 	+ "\"city\": \"Bangalore\","
				 	+ "\"cardIssuerCountryCode\": \"IND\""
				 + "},"
				 + "\"referenceNumber\": \"430000367618\","
				 + "\"name\": \"Mohammed Qasim\""
				 + "}";
	}
	
	@Test(groups = "visadirect")
	public void testWatchListInquiry() throws Exception {
	    String baseUri = "visadirect/";
	    String resourcePath = "watchlistscreening/v1/watchlistinquiry";
	    
	    CloseableHttpResponse response = this.abstractVisaAPIClient.doMutualAuthRequest(baseUri + resourcePath, "Watch List Inquiry Test", this.watchListInquiry, MethodTypes.POST, new HashMap<String, String>());
	    Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
	    response.close();
	}

}
